workbenchfx-core/src/main/java/com/dlsc/workbenchfx/view/ToolbarPresenter.java

```diff
@@ -130,6 +136,12 @@ public final void setupValueChangedListeners() {
     model.activeModuleProperty().addListener((observable, oldModule, newModule) -> {
       view.tabBar.setSelectedItem(newModule);
     });
+    model.getModules().addListener((ListChangeListener<WorkbenchModule>) c -> {
```

Comment:
I didn't test this, but as far as I can remember, we used to have issues in some cases because JavaFX triggers the change listeners first, passing the updated property to the change listener and only THEN changing the original property the change came from. This can result in concurrency issues, since you didn't take the change from the change listener but from the original property, if it hasn't been updated yet when this is executed.
Just to be safe: could you please refactor this to use an invalidation listener (like this: `(observable, oldModules, newModules) -> {`) and use the `newModules` instead of getting it via `model.getModules()`?

Reply:
Are you sure, that this problem occurs related to an `ObservableList`? 
`(observable, oldModules, newModules) -> {}` is a ChangeListener and the ObservableList only supports either a ListChangeListener (which is used right now) or a InvalidationListener.  

Also, at least the JavaDoc for the `ListChangeListener` says

> Called **after** a change has been made to an ObservableList.

I don't know if this is really the case but i never had any problems with that.

Reply:
Ah interesting, seems like I confused it with a `ChangeListener` then. In this case that's fine! :+1: thanks for pointing that out!